lib/ui/home/home_screen.dart

```diff
@@ -79,52 +79,79 @@ class HomeScreenState extends ConsumerState<HomeScreen> {
-          dotHeight: 8.0,
-          dotWidth: 8.0,
+  Widget _homeFooterWidget(List<SurveyModel> surveys) => HomeFooterWidget(
+        surveys: surveys,
+        pageController: _pageController,
+        onPageChangedCallback: (index) => _currentPageIndex.value = index,
+      );
+
+  Widget _pageIndicatorWidget(BuildContext context, int length) => Align(
+        alignment: Alignment.bottomLeft,
+        child: Padding(
+          padding: const EdgeInsets.only(bottom: 100.0),
+          child: SmoothPageIndicator(
+            controller: _pageController,
+            count: length,
+            effect: const WormEffect(
+              dotColor: Colors.white24,
+              activeDotColor: Colors.white,
+              dotHeight: 8.0,
+              dotWidth: 8.0,
+            ),
+          ),
         ),
       );
 
   Widget _homeWidget(BuildContext context) {
+    final surveys = ref.watch(surveysStream).value ?? [];
+    final focusedItemIndex = ref.watch(focusedItemIndexStream).value ?? 0;
+    _pageController = PageController(initialPage: focusedItemIndex);
     return Scaffold(
       key: _scaffoldKey,
+      backgroundColor: Colors.black,
       endDrawer: _sideMenu,
       body: Stack(
         children: [
-          PageView(
-            controller: _pageController,
-            onPageChanged: (index) => _currentPageIndex.value = index,
-            children: _homeBackground(ref.watch(surveysStream).value),
-          ),
-          SafeArea(
-            child: Container(
-              padding: const EdgeInsets.all(Dimensions.paddingMedium),
-              child: Column(
-                mainAxisAlignment: MainAxisAlignment.start,
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: [
-                  _homeHeaderWidget(),
-                  const Spacer(),
-                  _pageIndicatorWidget(
-                    context,
-                    ref.watch(surveysStream).value?.length ?? 0,
+          Visibility(
+            visible: surveys.isNotEmpty,
+            child: Stack(
+              children: [
+                _homeBackground(surveys, focusedItemIndex),
+                SafeArea(
+                  child: RefreshIndicator(
+                    color: Colors.black,
+                    backgroundColor: Colors.white,
+                    onRefresh: () => ref
+                        .read(homeViewModelProvider.notifier)
+                        .getSurveys(isRefresh: true),
+                    child: SingleChildScrollView(
+                      physics: const AlwaysScrollableScrollPhysics(),
+                      child: SizedBox(
+                        height: MediaQuery.of(context).size.height -
+                            MediaQuery.of(context).viewPadding.vertical,
+                        child: Container(
+                          padding: const EdgeInsets.all(
+                            Dimensions.paddingMedium,
+                          ),
```

Comment:
Please keep it in one line.

```suggestion
                          padding: const EdgeInsets.all(Dimensions.paddingMedium),
```

Reply:
@nmint8m we can't keep it in one line because this is out of max-line length (80) so it would look like this if we remove the trailing comma 🤔 
```
                          padding:
                              const EdgeInsets.all(Dimensions.paddingMedium),
```

Reply:
It depends on our monitor size. For me, this is not problem at all

Reply:
Yeah sure, no problem at all. Sorry, I didn't recognize that was a long line.